tests/test_metadata_integration.py

```diff
@@ -0,0 +1,269 @@
+
+from smartsim import Experiment
+from smartsim._core.config import CONFIG
+from smartsim.database.orchestrator import Orchestrator
+from smartsim.entity import Ensemble, Model
+from smartsim.settings import RunSettings
+
+
+class TestMetadataDirectoryIntegration:
+    """Integration tests for metadata directory creation across the SmartSim workflow"""
+
+    def test_experiment_creates_correct_metadata_directory_structure_model_only(self):
+        """Test that launching only models creates the correct directory structure"""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            exp = Experiment("test_metadata_model", exp_path=temp_dir, launcher="local")
+
+            # Create a simple model
+            model = exp.create_model(
+                "test_model", run_settings=exp.create_run_settings("echo", ["hello"])
+            )
+
+            # Start and wait for completion
+            exp.start(model, block=True)
+
+            # Verify directory structure
+            smartsim_dir = pathlib.Path(temp_dir) / CONFIG.smartsim_base_dir
+            metadata_dir = smartsim_dir / "metadata"
+
+            assert metadata_dir.exists(), "Metadata directory should exist"
+
+            # Check for run-specific subdirectory
+            run_dirs = [
+                d
+                for d in metadata_dir.iterdir()
+                if d.is_dir() and d.name.startswith("run_")
+            ]
+            assert (
+                len(run_dirs) == 1
+            ), f"Should have exactly one run directory, found: {run_dirs}"
+
+            run_dir = run_dirs[0]
+
+            # Check for entity-specific subdirectories with entity names
+            model_dir = run_dir / "model" / "test_model"
+            ensemble_dir = run_dir / "ensemble"
+            database_dir = run_dir / "database"
+
+            assert (
+                model_dir.exists()
+            ), f"Model metadata directory should exist: {model_dir}"
+            assert (
+                not ensemble_dir.exists()
+            ), f"Ensemble metadata directory should not exist: {ensemble_dir}"
+            assert (
+                not database_dir.exists()
+            ), f"Database metadata directory should not exist: {database_dir}"
+
+    def test_experiment_creates_correct_metadata_directory_structure_ensemble_only(
+        self,
+    ):
+        """Test that launching only ensembles creates the correct directory structure"""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            exp = Experiment(
+                "test_metadata_ensemble", exp_path=temp_dir, launcher="local"
+            )
+
+            # Create an ensemble
+            ensemble = exp.create_ensemble(
+                "test_ensemble",
+                run_settings=exp.create_run_settings("echo", ["world"]),
+                replicas=2,
+            )
+
+            # Start and wait for completion
+            exp.start(ensemble, block=False)
+            exp.poll(interval=1)
```

Comment:
Style nit: equivalent to `exp.start(ensemble, block=True)`? Any reason for the separation?